src/utils/getQueryStringFromUrl.ts

```diff
@@ -0,0 +1,9 @@
+import { NextRouter } from 'next/router';
+
+export const getQueryStringFromUrl = (router: NextRouter) => {
+    const qQueryParamValue = router?.query?.q as string;
+    if (qQueryParamValue) {
+        const searchStringArr = qQueryParamValue.split(':');
+        return searchStringArr[1];
+    }
```

Comment:
You are expecting a string to be returned from this function. Empty string should be returned if query is not present. Then you won't need nullish coalescing here `(searchText ?? '')` and also won't need type casting here `const searchString = getQueryStringFromUrl(router) as string;`